BibBuddy/app/src/main/java/de/bibbuddy/ExportBibTex.java

```diff
@@ -0,0 +1,175 @@
+   *
+   * @param shelfId id of the shelf
+   * @param bookDao object of the BookDao class
+   *                Depends on the used fragment.
+   *                Can be accessed through BookFragment or LibraryModel.
+   * @param noteDao object of the class NoteDao
+   *                Depends on the used fragment.
+   *                Can be accessed through BookFragment, LibraryModel or
+   *                NoteModel.
+   */
+  public String getBibFormatBook(Long shelfId, BookDao bookDao, NoteDao noteDao) {
+
+    List<Long> bookIdsCurrentShelf =
+        bookDao.getAllBookIdsForShelf(shelfId);
+    String bibFormat = "";
+
+    // for each book in the current shelf
+    for (int i = 0; i < bookIdsCurrentShelf.size(); i++) {
+      Book book;
+      book = bookDao.findById(bookIdsCurrentShelf.get(i));
+      Long bookId = bookIdsCurrentShelf.get(i);
+
+      bibFormat =  "@book{" + getBibKey(book)
+          + "isbn={" + book.getIsbn() + "}," + '\n'
+          + getBibAuthorNames(bookId, bookDao)
+          + "title={" + book.getTitle() + "}," + '\n'
+          + "subtitle={" + book.getSubtitle() + "}," + '\n'
+          + "volume={" + book.getVolume() + "}," + '\n'
+          + "publisher={" + book.getPublisher() + "}," + '\n'
+          + "edition={" + book.getEdition() + "}," + '\n'
+          + getBibNotes(book, noteDao)
+          + "year=" + book.getPubYear() + '\n' + "}" + '\n' + '\n';
+
+    }
+    return bibFormat;
+  }
+
+  private String getBibKey(Book book) {
+    // remove whitespaces from book's title
+    String bookTitle = book.getTitle().replaceAll("\\s+", "");
+    return bookTitle + "," + '\n';
+  }
+
+  private String getBibNotes(Book book, NoteDao noteDao) {
+    List<Long> notesList = noteDao.getAllNoteIdsForBook(book.getId());
+    String allNotes = "";
+
+    if (!notesList.isEmpty()) {
+      for (int k = 0; k < notesList.size(); k++) {
+        String bookTextNotes = noteDao.findTextById(notesList.get(k));
+        allNotes +=  "annote={" + bookTextNotes + "}," + '\n';
+      }
+    }
+
+    return allNotes;
+  }
+
+  private String getBibAuthorNames(Long bookId, BookDao bookDao) {
+    Book book;
+    book = bookDao.findById(bookId);
+
+    List<Author> authorsList = new ArrayList<>();
+    authorsList = bookDao.getAllAuthorsForBook(book.getId());
+    String authorNames = "";
+
+    if (authorsList.size() > 1) {
+
+      for (int i = 0; i < authorsList.size(); i++) {
+        authorNames = authorNames
+            + authorsList.get(i).getFirstName()
+            + " " + authorsList.get(i).getLastName();
+        if (i < authorsList.size()) {
+          authorNames = authorNames + " and ";
+        }
+      }
+
+    } else {
+
+      try {
+        authorNames = authorsList.get(0).getFirstName()
+            + " " + authorsList.get(0).getLastName();
+      } catch (Exception e) {
+        authorNames = "";
+      }
+
+    }
+    return "author={" + authorNames + "}," + '\n';
+  }
```

Comment:
```suggestion
  private String getBibAuthorNames(Long bookId, BookDao bookDao) {
    Book book = bookDao.findById(bookId);

    List<Author> authorsList = bookDao.getAllAuthorsForBook(book.getId());
    StringBuilder authorNames = new StringBuilder();

    for (int i = 0; i < authorsList.size(); i++) {
        authorNames.append(authorsList.get(i).getFirstName()).append(" ")
            .append(authorsList.get(i).getLastName());
        
        if (i < authorsList.size() && i != 0) {
          authorNames.append(" and ");
        }
    }

    return "author={" + authorNames.toString() + "}," + '\n';
  }
```

Reply:
This solution was already reviewed. Everyone has a different solution. Since we have already reviewed it in #89 I would not change it anymore.

Reply:
Ok then we leave it as it is. But could you just change the initialization of `authorsList` to `List<Author> authorsList = bookDao.getAllAuthorsForBook(book.getId());`